rtgui/edit.cc

```diff
@@ -747,7 +747,294 @@ void Rectangle::drawToMOChannel(Cairo::RefPtr<Cairo::Context> &cr, unsigned shor
+               Cairo::RefPtr<Cairo::ImageSurface> *insensitive,
+               enum DrivenPoint drivenPoint) :
+    drivenPoint(drivenPoint)
+{
+    if (normal) {
+        normalImg = *normal;
+    }
+
+    if (prelight) {
+        prelightImg = *prelight;
+    }
+
+    if (active) {
+        activeImg = *active;
+    }
+
+    if (dragged) {
+        draggedImg = *active;
+    }
+
+    if (insensitive) {
+        insensitiveImg = *insensitive;
+    }
+}
+
+OPIcon::OPIcon(Glib::ustring normalImage, Glib::ustring activeImage, Glib::ustring prelightImage,
+               Glib::ustring  draggedImage, Glib::ustring insensitiveImage, enum DrivenPoint drivenPoint) : drivenPoint(drivenPoint)
+{
+    if (!normalImage.empty()) {
+        normalImg = Cairo::ImageSurface::create_from_png( RTImage::findIconAbsolutePath(normalImage) );
+    }
+
+    if (!prelightImage.empty()) {
+        prelightImg = Cairo::ImageSurface::create_from_png( RTImage::findIconAbsolutePath(prelightImage) );
+    }
+
+    if (!activeImage.empty()) {
+        activeImg = Cairo::ImageSurface::create_from_png( RTImage::findIconAbsolutePath(activeImage) );
+    }
+
+    if (!draggedImage.empty()) {
+        draggedImg = Cairo::ImageSurface::create_from_png( RTImage::findIconAbsolutePath(draggedImage) );
+    }
+
+    if (!insensitiveImage.empty()) {
+        insensitiveImg = Cairo::ImageSurface::create_from_png( RTImage::findIconAbsolutePath(insensitiveImage) );
+    }
+}
+
+Cairo::RefPtr<Cairo::ImageSurface> OPIcon::getNormalImg()
+{
+    return normalImg;
+}
+Cairo::RefPtr<Cairo::ImageSurface> OPIcon::getPrelightImg()
+{
+    return prelightImg;
+}
+Cairo::RefPtr<Cairo::ImageSurface> OPIcon::getActiveImg()
+{
+    return activeImg;
+}
+Cairo::RefPtr<Cairo::ImageSurface> OPIcon::getDraggedImg()
+{
+    return draggedImg;
+}
+Cairo::RefPtr<Cairo::ImageSurface> OPIcon::getInsensitiveImg()
+{
+    return insensitiveImg;
+}
+
+void OPIcon::drawImage(Cairo::RefPtr<Cairo::ImageSurface> img,
+                       Cairo::RefPtr<Cairo::Context> &cr, ObjectMOBuffer *objectBuffer,
+                       EditCoordSystem &coordSystem)
+{
+    int imgW = img->get_width();
+    int imgH = img->get_height();
+
+    rtengine::Coord pos;
+
+    if (datum == IMAGE) {
+        coordSystem.imageCoordToScreen(position.x, position.y, pos.x, pos.y);
+    } else if (datum == CLICKED_POINT) {
+        pos = position + objectBuffer->getDataProvider()->posScreen;
+    } else if (datum == CURSOR)
+        pos = position + objectBuffer->getDataProvider()->posScreen
+              + objectBuffer->getDataProvider()->deltaScreen;
+
+    rtengine::Coord tl, br; // Coordinate of the rectangle in the CropBuffer coordinate system
+    drivenPointToRectangle(pos, tl, br, imgW, imgH);
+
+    cr->set_source(img, tl.x, tl.y);
+    cr->set_line_width(0.);
+    cr->rectangle(tl.x, tl.y, imgW, imgH);
+    cr->fill();
+}
+
+void OPIcon::drawMOImage(Cairo::RefPtr<Cairo::ImageSurface> img, Cairo::RefPtr<Cairo::Context> &cr,
```

Comment:
Same as above. Also `const Cairo::RefPtr<Cairo::Context>&` seems appropriate. (The `const` that woudl be prevent the context from being modified would be `Cairo::RefPtr<const Cairo::Context>`.)